packages/tools/src/tools/annotation/VideoRedactionTool.ts

```diff
@@ -0,0 +1,935 @@
+      return true;
+    }
+  };
+
+  toolSelectedCallback = (evt, annotation, interactionType = 'mouse') => {
+    const eventData = evt.detail;
+    const { element } = eventData;
+
+    const { data } = annotation;
+
+    data.active = true;
+
+    const viewportUIDsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName(),
+      false // TODO_JAMES: Just trying to bypass the vtk specific stuff as fast as possible.
+    );
+
+    this.editData = {
+      annotation,
+      viewportUIDsToRender,
+      movingTextBox: false,
+    };
+
+    this._activateModify(element);
+
+    hideElementCursor(element);
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine } = enabledElement;
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      viewportUIDsToRender
+    );
+
+    evt.preventDefault();
+  };
+
+  handleSelectedCallback = (
+    evt,
+    annotation,
+    handle,
+    interactionType = 'mouse'
+  ) => {
+    const eventData = evt.detail;
+    const { element } = eventData;
+    const { data } = annotation;
+
+    data.active = true;
+
+    let movingTextBox = false;
+    let handleIndex;
+
+    if (handle.worldPosition) {
+      movingTextBox = true;
+    } else {
+      handleIndex = data.handles.points.findIndex((p) => p === handle);
+    }
+
+    // Find viewports to render on drag.
+    const viewportUIDsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName(),
+      false // TODO_JAMES: Just trying to bypass the vtk specific stuff as fast as possible.
+    );
+
+    this.editData = {
+      annotation,
+      viewportUIDsToRender,
+      handleIndex,
+      movingTextBox,
+    };
+    this._activateModify(element);
+
+    hideElementCursor(element);
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine } = enabledElement;
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      viewportUIDsToRender
+    );
+
+    evt.preventDefault();
+  };
+
+  _mouseUpCallback = (evt) => {
+    const eventData = evt.detail;
+    const { element } = eventData;
+
+    const { annotation, viewportUIDsToRender, newAnnotation, hasMoved } =
+      this.editData;
+    const { data } = annotation;
+
+    if (newAnnotation && !hasMoved) {
+      return;
+    }
+
+    data.active = false;
+    data.handles.activeHandleIndex = null;
+
+    this._deactivateModify(element);
+    this._deactivateDraw(element);
+
+    resetElementCursor(element);
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine } = enabledElement;
+
+    this.editData = null;
+    this.isDrawing = false;
+
+    if (
+      this.isHandleOutsideImage &&
+      this.configuration.preventHandleOutsideImage
+    ) {
+      removeAnnotation(annotation.annotationUID);
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      viewportUIDsToRender
+    );
+  };
+
```

Comment:
I feel like we should extend the RectangleROI Tool and just override the renderAnnotation. Similar to how we do. There are a lot of repetitive code here

packages/tools/src/tools/segmentation/RectangleROIThresholdTool.ts

Reply:
Agreed, but that belongs in the tool section for the video.  Right now it is fine as an example that is intended to still evolve.